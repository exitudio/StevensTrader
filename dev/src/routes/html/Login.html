<html>
    
    <head>   
		<title>Welcome to Stevens Trader System - Admin Login </title>
        
        <style type="text/css">
			body {text-align:center;
				  color: #000;}
			.MndMrk {color: red;}
		</style>
	</head>

    <body>
		<form id="LinFom" action="LinFom.php" method="post">
			<div id="LinDiv">
			
				<h2>Stevens Trader System - Admin Login</h2>
				<table width="600" border="0" align="center" cellspacing="10">
				  <tr>
					<td width="200" align="right"><label for="LinUID">User Name :<span class="MndMrk">*</span></label></td>
					<td><input type="text" name="LinUID" id="LinUID" size="50"/></td>
				  </tr>
				  <tr>
					<td width="200" align="right"><label for="LinPwd">Password :<span class="MndMrk">*</span></label></td>
					<td><input type="password" name="LinPwd" id="LinPwd" size="50"/></td>
				  </tr>
				</table>
				<br>
				<button id="LinBtn"> Login </button>
			   
			</div>
		</form>
		<div id="Ack"></div>
		
		<script type="text/javascript" src="JqueryScript-v2.1.3.js"></script>
		<script type="text/javascript" src="LinFom.js"></script>
	</body>
	
</html>
