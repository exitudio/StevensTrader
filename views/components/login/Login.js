import React from 'react';
import ReactDom from 'react-dom';
// import SignupScss from '../../scss/Signup.scss';
import {loginAction} from '../../redux/actions/accountActions';
import store from '../../redux/store';

const EMAIL = "email";
const PASSWORD = "password";
export default class Login extends React.Component{

    constructor(props){
        super(props);
        this.state = {
            isEmailCorrect:true, 
            isPasswordCorrect:true, 
        };
    }
    handleSubmit = (event)=>{
        console.dir(event);
        event.preventDefault();
        if(
            this.state.isEmailCorrect &&
            this.state.isPasswordCorrect
        ){
            //submit
            store.dispatch( loginAction({
                email:this.state.email,
                password:this.state.password
            }) );
        }
    }

    onChange = (event)=>{
        var state = {};
        var value = event.target.value.trim();
       
        if(event.target.name===EMAIL){
            if(this.validateEmail(value)){
                state.isEmailCorrect = true;
            }else{
                state.isEmailCorrect = false;
            }
        }else if(event.target.name===PASSWORD){
            if(value===""){
                state.isPasswordCorrect = false;
            }else{
                state.isPasswordCorrect = true;
            }
        }
        state[event.target.name] = value;
        this.setState(state);
    }

    validateEmail(email){
        return /^\"?[\w-_\.]*\"?@stevens\.edu$/.test(email);        
    }

    checkStateClass(isCorrect){
        if(!isCorrect){
            return "wrong";
        }
        return "";
    }
    
    render(){
        return(
            <div className="signup-container">
                <form onSubmit={this.handleSubmit}>
                    <h1 >Log In</h1>
                    <hr/>

                    <h3>Email</h3>
                    <input type="text"      name={EMAIL} onChange={this.onChange} className={`${this.checkStateClass(this.state.isEmailCorrect)}`}/>
                    <div className={`warning ${this.checkStateClass(this.state.isEmailCorrect)}`}>Email should be Stevens's email.</div>

                    <h3>Password</h3>
                    <input type="password"  name={PASSWORD} onChange={this.onChange} className={`${this.checkStateClass(this.state.isPasswordCorrect)}`}/>
                    <div className={`warning ${this.checkStateClass(this.state.isPasswordCorrect)}`}>Password can not be empty.</div>

                    <input type="submit" value="SUBMIT" id="submit-button" className="button"/>
                </form>
            </div>
        );
    }
}