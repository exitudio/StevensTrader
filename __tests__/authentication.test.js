import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import Login from '../views/components/login/Login'

describe('Login (Snapshot)', () => {
  it('Login renders', () => {
    const component = renderer.create(<Login />)
    const json = component.toJSON()
    expect(json).toMatchSnapshot()
  })
})

describe('---- Item Testing -------', () => {
        it('Login warning',()=>{
            console.log('----------------------------')
            const login = shallow(<Login />)
            login.setState({isEmailCorrect:true, 
                    isPasswordCorrect:true, })

            console.log(login.find('.warning'))
            expect(login.find('.warning')).toEqual('Hello world')


            // isEmailCorrect:true, 
            //         isPasswordCorrect:true, 
        })
    })