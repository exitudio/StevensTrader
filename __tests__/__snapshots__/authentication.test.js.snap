// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login (Snapshot) Login renders 1`] = `
<div
  className="signup-container"
>
  <form
    onSubmit={[Function]}
  >
    <h1>
      Log In
    </h1>
    <hr />
    <h3>
      Email
    </h3>
    <input
      className=""
      name="email"
      onChange={[Function]}
      type="text"
    />
    <div
      className="warning "
    >
      Email should be Stevens's email.
    </div>
    <h3>
      Password
    </h3>
    <input
      className=""
      name="password"
      onChange={[Function]}
      type="password"
    />
    <div
      className="warning "
    >
      Password can not be empty.
    </div>
    <input
      className="button"
      id="submit-button"
      type="submit"
      value="SUBMIT"
    />
  </form>
</div>
`;
